spring-boot-security-taglibs/src/main/java/org/baeldung/security/ApplicationConfig.java

```diff
@@ -0,0 +1,40 @@
+package org.baeldung.security;
+
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@SpringBootApplication
+@Configuration
+@EnableWebSecurity
+public class ApplicationConfig extends WebSecurityConfigurerAdapter {
+    public static final String DEFAULT_PASSWORD = "password";
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        PasswordEncoder encoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
```

Comment:
I applaud your extra efforts to teach good security practices.

It is my opinion, however, that this extra work is unnecessary because I think that the typical reader sees that you are using inMemoryAuthentication.

If you take a look at the [Java Docs for InMemoryUserDetailsManager](https://docs.spring.io/spring-security/site/docs/4.2.6.RELEASE/apidocs/org/springframework/security/provisioning/InMemoryUserDetailsManager.html), it says:

> Mainly intended for testing and demonstration purposes

And I don't believe that folks will use this part of your code for the long-term basis of their app.

Further, since the password is in-memory, there is no password at rest to be stolen, and so the encoding used is immaterial. If they were to steal it from the string pool, they have it already anyway in the way of `DEFAULT_PASSWORD`.

Again, I like you security awareness, but it is just noise in this article.

Keep in mind that the very authors of the library you are demonstrating don't add this complexity to their intro examples.